plugins/modules/public_cloud_interface_instance.py

```diff
@@ -0,0 +1,117 @@
+
+description:
+    - This module attach or detach an interface of an instance on OVH public Cloud.
+
+requirements:
+    - ovh >= 0.5.0
+
+options:
+    service_name:
+        required: true
+        description: The service_name
+    instance_id:
+        required: true
+        description: The instance uuid
+    state:
+        required: false
+        default: present
+        choices: ['present','absent']
+        description: Indicate the desired state of the interface
+    interface_ip:
+        required: true
+        description: The fixed IP to set to the interface
+    interface_network_id:
+        required: true
+        description: The network id to attache the interface to
+'''
+
+EXAMPLES = r'''
+  - name: Create vrack interface
+  synthesio.ovh.public_cloud_interface_instance:
+    service_name: "{{ service_name }}"
+    instance_id: "{{ instance_id }}"
+    interface_ip: "{{ network_vrack.ip }}"
+    interface_network_id: "{{ network_vrack.network_id }}"
+  delegate_to: localhost
+  register: interface_metadata
+
+'''
+
+RETURN = r''' # '''
+
+from ansible_collections.synthesio.ovh.plugins.module_utils.ovh import ovh_api_connect, ovh_argument_spec
+
+try:
+    from ovh.exceptions import APIError
+
+    HAS_OVH = True
+except ImportError:
+    HAS_OVH = False
+
+
+def run_module():
+    module_args = ovh_argument_spec()
+    module_args.update(dict(
+        service_name=dict(required=True),
+        instance_id=dict(required=True),
+        state=dict(choices=['present', 'absent'], default='present'),
+        interface_ip=dict(required=True),
+        interface_network_id=dict(required=True)
+    ))
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
+    )
+    client = ovh_api_connect(module)
+
+    service_name = module.params['service_name']
+    instance_id = module.params['instance_id']
+    state = module.params['state']
+    interface_ip = module.params['interface_ip']
+    interface_network_id = module.params['interface_network_id']
+
+    if module.check_mode:
+        module.exit_json(msg="Ensure interface {} on {} is {} on instance id {} - (dry run mode)"
+                         .format(interface_ip, interface_network_id, state, instance_id),
+                         changed=True)
+
+    if state == 'absent':
+        # Need to get the interface id (via /cloud/project/{serviceName}/instance/{instanceId}/interface).
+        # How to manage multiple interfaces ?
+        module.fail_json(msg="Removing an interface is not yet implemented")
+    if state == 'present':
+        try:
+            result = client.post('/cloud/project/%s/instance/%s/interface' % (service_name, instance_id),
+                                 ip=interface_ip,
+                                 networkId=interface_network_id)
+            module.exit_json(
+                changed=True,
+                msg="Interface has been attached to instance id {}".format(
```

Comment:
It may be clearer to use the service name instead of the instance id?
I can't remember the context, if the service name is the project or the hostname

Reply:
Yes, `service_name` is the public cloud project ID. Naming is bad, but this is the name in OVH API